Code/Framework/AzCore/AzCore/Memory/AllocatorManager.cpp

```diff
@@ -14,394 +14,710 @@
+            return AZ::StringFunc::Equal(trackingConfig.m_allocatorName, allocatorName);
+        };
+        return AZStd::erase_if(m_allocatorTrackingConfigs, FindAllocatorByName) != 0;
+    }
 
-    // Build a mapping of original allocator sources to their allocators
-    for (int i = 0; i < allocatorCount; ++i)
+    void AllocatorManager::EnterProfilingMode()
     {
-        IAllocator* allocator = GetAllocator(i);
-        allocatedBytes += allocator->NumAllocatedBytes();
-        capacityBytes += allocator->Capacity();   
-           
-        if (outStats)
+        AZStd::lock_guard<AZStd::mutex> lock(m_allocatorListMutex);
+        for (int i = 0; i < m_numAllocators; ++i)
         {
-            outStats->emplace(outStats->end(),
-                allocator->GetName(),
-                allocator->NumAllocatedBytes(),
-                allocator->Capacity());
+            m_allocators[i]->SetProfilingActive(true);
         }
     }
-}
 
-//=========================================================================
-// MemoryBreak
-// [2/24/2011]
-//=========================================================================
-AllocatorManager::MemoryBreak::MemoryBreak()
-{
-    addressStart = nullptr;
-    addressEnd = nullptr;
-    byteSize = 0;
-    alignment = static_cast<size_t>(0xffffffff);
-    name = nullptr;
-
-    fileName = nullptr;
-    lineNum = -1;
-}
-
-//=========================================================================
-// SetMemoryBreak
-// [2/24/2011]
-//=========================================================================
-void
-AllocatorManager::SetMemoryBreak(int slot, const MemoryBreak& mb)
-{
-    AZ_Assert(slot < MaxNumMemoryBreaks, "Invalid slot index");
-    m_activeBreaks |= 1 << slot;
-    m_memoryBreak[slot] = mb;
-}
-
-//=========================================================================
-// ResetMemoryBreak
-// [2/24/2011]
-//=========================================================================
-void
-AllocatorManager::ResetMemoryBreak(int slot)
-{
-    if (slot == -1)
+    void AllocatorManager::ExitProfilingMode()
     {
-        m_activeBreaks = 0;
+        AZStd::lock_guard<AZStd::mutex> lock(m_allocatorListMutex);
+        for (int i = 0; i < m_numAllocators; ++i)
+        {
+            m_allocators[i]->SetProfilingActive(false);
+        }
     }
-    else
+
+    void AllocatorManager::DumpAllocators()
     {
-        AZ_Assert(slot < MaxNumMemoryBreaks, "Invalid slot index");
-        m_activeBreaks &= ~(1 << slot);
+        static constexpr const char* TAG = "mem";
+
+        AZStd::lock_guard<AZStd::mutex> lock(m_allocatorListMutex);
+        size_t totalUsedBytes = 0;
+        size_t totalReservedBytes = 0;
+        size_t totalConsumedBytes = 0;
+
+        memset(m_dumpInfo, 0, sizeof(m_dumpInfo));
+
+        AZ_Printf(TAG, "%d allocators active\n", m_numAllocators);
+        AZ_Printf(TAG, "Index,Name,Used kb,Reserved kb,Consumed kb\n");
```

Comment:
The "mem" tag and the first line printed ("%d allocators active") seems like they prevent this from being a well-formed CSV. Consider removing them, making them optional, or at least moving the allocators active line to the end so that it doesn't mess up the column headers.

Reply:
I am replacing the "mem" tag with the new NoWindow constant to skip printing the window tag.

I am also moving the printing of the amount of allocators active to the end of the DumpAllocators function.